coordinator/src/position/mod.rs

```diff
@@ -1 +1,92 @@
 pub mod models;
+
+use crate::db;
+use crate::node::Node;
+use crate::node::TradeAction;
+use crate::position::models::Position;
+use crate::position::models::PositionState;
+use anyhow::Context;
+use anyhow::Result;
+use std::sync::Arc;
+use time::OffsetDateTime;
+use tokio::task::spawn_blocking;
+use trade::bitmex_client::BitmexClient;
+
+pub async fn sync_positions(node: Arc<Node>) -> Result<()> {
```

Comment:
🤔 thy do you think so? the idea is to have multiple `await` points to minimise idle time on the thread.
maybe let's wait for the update to this PR, or let's have a chat about it!

Reply:
In https://github.com/get10101/10101/commit/bf7acb2aed02f84d28d353cddfcc95da25314634 I'm making `close_position` sync (because it's all blocking code), which removes one of the `await` points.

You've added two `await` points which do make sense if we want to keep this function `async` (but which obviously don't need to be there if we make it sync).

Finally there is `get_quote`, which is definitely supposed to be `async` and which I'm proposing to call via `runtime.block_on`.

The way I'm seeing this working is:

```rust
tokio::spawn({
    async move {
        loop {
            tokio::time::sleep(POSITION_SYNC_INTERVAL).await;
            spawn_blocking(sync_positions(node)).await;
        }
    }
});

fn sync_positions(node: Node) {
    let positions = get_open_positions();

    for position in positions {
        let channel_id = decide_trade_action(node, position.trader);

        let closing_price = runtime.block_on(get_quote(&position.expiry_timestamp).await);

        node.close_position(position, closing_price, channel_id);
    }
}
```

I'm obviously not an expert, but my understanding is that this is better because this function is actually primarily blocking, with just one spot where async is actually necessary/desirable.